pkg/notifications/client.go

```diff
@@ -71,6 +69,28 @@ func MapRepositoryResponse(importedRepo api.RepositoryResponse) repositories.Rep
 	}
 }
 
+func MapRepositoryConfig(importedRepo models.RepositoryConfiguration) repositories.Repositories {
+	packageCount := int64(importedRepo.Repository.PackageCount)
+	failedIntrospectionsCount := int64(importedRepo.Repository.FailedIntrospectionsCount)
+
+	return repositories.Repositories{
+		Name:                         importedRepo.Name,
+		URL:                          importedRepo.Repository.URL,
+		UUID:                         importedRepo.UUID,
+		DistributionVersions:         importedRepo.Versions,
+		FailedIntrospectionsCount:    &failedIntrospectionsCount,         // Optional but defaults to 0
+		PackageCount:                 &packageCount,                      // Optional but defaults to 0
+		MetadataVerification:         &importedRepo.MetadataVerification, // Optional but defaults to false
+		DistributionArch:             SetEmptyToNil(importedRepo.Arch),   // Below are all optional, we need to set them as nil if empty for the cloud schema
+		GPGKey:                       SetEmptyToNil(importedRepo.GpgKey),
+		LastIntrospectionError:       SetEmptyToNil(*importedRepo.Repository.LastIntrospectionError),
+		LastIntrospectionTime:        SetEmptyToNil(importedRepo.Repository.LastIntrospectionTime.String()),
+		LastSuccessIntrospectionTime: SetEmptyToNil(importedRepo.Repository.LastIntrospectionSuccessTime.String()),
+		LastUpdateIntrospectionTime:  SetEmptyToNil(importedRepo.Repository.LastIntrospectionUpdateTime.String()),
```

Comment:
These 3 lines are assuming that the Time is not nil, but in our test cases they are actually nil, and nil.String() blows up with a panic, this is your test failure

Reply:
OH! nice catch. Thank you.